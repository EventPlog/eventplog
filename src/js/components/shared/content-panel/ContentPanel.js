import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  background: white;
  padding: 1rem;
  box-shadow: 1px 2px 4px ${props => lighten(0.45, props.theme.activeLink)};
    
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    border-bottom: 1px solid ${props => lighten(0.2, props.theme.blue)};
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 0.01rem;
    margin: 0 1rem;
    padding-bottom: 5px;
    
    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    margin: 2rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h5 className="content-header">{ title }</h5>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel