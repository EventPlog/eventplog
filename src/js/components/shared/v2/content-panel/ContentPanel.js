import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    margin: 0;
  }

    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    // margin: 2rem 1rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h4 className="content-header">{ title }</h4>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel
