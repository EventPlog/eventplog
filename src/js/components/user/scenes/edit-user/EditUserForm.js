import React from 'react';
import { Checkbox, Form, Select, Message } from 'semantic-ui-react';
import styled from 'styled-components';
import Input from 'js/components/shared/input'
import TextArea from 'js/components/shared/text-area'
import Button from 'js/components/shared/button'
import ContentPanel from 'js/components/shared/content-panel'
import Loading from "js/components/shared/loading";
import { media } from 'js/styles/mixins'
import color from 'js/styles/theme/variables'
import { genEventLink, titleize } from 'js/utils'
import colorOptions from 'js/utils/colorOptions'

const StyledEditUserForm = styled.div`
  .content-panel {
    width: 100%;
  }
  
  .content-header {
    ${
      media.phone`
        margin: 0;
        margin-bottom: 2rem;
      `
    }
  }
  
  .content-body {
    margin-left: 0;
    margin-right: 0;  
  }
  
  .ui.form {
    max-width: 400px;
    margin: auto; 
    
    .check-user {
       margin-bottom: 2rem;
    }
  }
  
  button + button {
    margin-left: 1rem;
  }
  
  .space-top {
    margin-top: 2rem;
  }
`


const genderOptions = [
  { key: 'm', text: 'Male', value: 'male' },
  { key: 'f', text: 'Female', value: 'female' },
]

const emptyUser = {
  first_name: '',
  last_name: '',
  email: '',
  gender: 'Male'
}

const EditUserForm = ({
  user = emptyUser,
  event = {},
  check_in_user,
  handleChange,
  handleStateChange,
  handleSubmit,
  success,
  error,
  loading,
}) => {
  if (loading) return <Loading />
  const title = <a href={`${window.location.origin}${genEventLink(event, event.community)}?utm_source=check_in_form`}>{event.title}</a>
  return (
    <StyledEditUserForm className="">
      <div className="app-container">
        <ContentPanel title={<p>Edit Your Personal Information</p>}>
          <Form loading={loading} success={success} error={error}>
            <Message
              success
              header='Success!'
              content={success}
            />
            <Message
              error
              header='Error!'
              content={error}
            />

            <Form.Field>
              <label>First Name</label>
              <Input name="first_name"
                     value={titleize(user.first_name)}
                     placeholder='Ciroma'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Last Name</label>
              <Input name="last_name"
                     value={titleize(user.last_name)}
                     placeholder='Chukwuma'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Gender</label>
              <Select onChange={(e, attr) => handleChange('gender', attr.value)}
                      value={user.gender}
                      defaultValue={'male'}
                      placeholder='Gender' options={genderOptions} />
            </Form.Field>

            <Form.Field>
              <label>Email</label>
              <Input name="email"
                     disabled
                     value={user.email}
                     placeholder='ciroma@chukwuma.com'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Bio</label>
              <TextArea name="bio"
                        value={user.bio}
                        placeholder='A bit about yourself'
                        maxLength={1000}
                        rows={5}
                        onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Preferred link color</label>
                <Select options={colorOptions}
                        name="brand_color"
                        value={user.brand_color}
                        defaultValue={color.activeLink}
                        onChange={(e, attr) => handleChange(attr.name, attr.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Occupation</label>
              <Input name="occupation"
                     value={user.occupation}
                     placeholder='Content Creator'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Facebook</label>
              <Input name="facebook_profile"
                     value={user.facebook_profile}
                     placeholder='ciroma@chukwuma.com'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Twitter</label>
              <Input name="twitter_profile"
                     value={user.twitter_profile}
                     placeholder='ciroma@chukwuma.com'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>LinkedIn</label>
              <Input name="linkedin_profile"
                     value={user.linkedin_profile}
                     placeholder='linkedin.com/something'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Github</label>
              <Input name="github_profile"
                     value={user.github_profile}
                     placeholder='github.com/you'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Form.Field>
              <label>Personal website/blog</label>
              <Input name="site_link"
                     value={user.site_link}
                     placeholder='ciroma@chukwuma.com'
                     onChange={({target}) => handleChange(target.name, target.value)}/>
            </Form.Field>

            <Button inverted
                    type='submit'
                    disabled={!user.email}
                    className="space-top"
                    onClick={handleSubmit}>
              Save
            </Button>
          </Form>
        </ContentPanel>
      </div>
    </StyledEditUserForm>
  )
}

export default EditUserForm;