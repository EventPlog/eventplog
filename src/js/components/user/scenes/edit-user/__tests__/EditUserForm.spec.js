import React from 'react';
import EditUserForm from '../EditUserForm';
import ContentPanel from 'js/components/shared/content-panel'
import { shallow } from 'enzyme';

describe('EditUserForm', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <EditUserForm  /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(ContentPanel).length).toEqual(1);
  })
});