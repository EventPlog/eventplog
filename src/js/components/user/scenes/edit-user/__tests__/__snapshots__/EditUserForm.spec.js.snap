// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserForm should render correctly 1`] = `
<styled.div
  className=""
>
  <div
    className="app-container"
  >
    <ContentPanel
      title={
        <p>
          Edit Your Personal Information
        </p>
      }
    >
      <Form
        as="form"
      >
        <Message
          header="Success!"
          success={true}
        />
        <Message
          error={true}
          header="Error!"
        />
        <FormField>
          <label>
            First Name
          </label>
          <Styled(Input)
            name="first_name"
            onChange={[Function]}
            placeholder="Ciroma"
            value=""
          />
        </FormField>
        <FormField>
          <label>
            Last Name
          </label>
          <Styled(Input)
            name="last_name"
            onChange={[Function]}
            placeholder="Chukwuma"
            value=""
          />
        </FormField>
        <FormField>
          <label>
            Gender
          </label>
          <Select
            defaultValue="male"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "key": "m",
                  "text": "Male",
                  "value": "male",
                },
                Object {
                  "key": "f",
                  "text": "Female",
                  "value": "female",
                },
              ]
            }
            placeholder="Gender"
            value="Male"
          />
        </FormField>
        <FormField>
          <label>
            Email
          </label>
          <Styled(Input)
            name="email"
            onChange={[Function]}
            placeholder="ciroma@chukwuma.com"
            value=""
          />
        </FormField>
        <FormField>
          <label>
            Bio
          </label>
          <Styled(Component)
            maxLength={70}
            name="bio"
            onChange={[Function]}
            placeholder="A bit about yourself"
          />
        </FormField>
        <FormField>
          <label>
            Preferred link color
          </label>
          <Select
            defaultValue="#9448BC"
            name="brand_color"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "icon": "point purple",
                  "key": "default",
                  "text": "default",
                  "value": "#9448BC",
                },
                Object {
                  "icon": "point blue",
                  "key": "blue",
                  "text": "blue",
                  "value": "#4968B7",
                },
                Object {
                  "icon": "point red",
                  "key": "red",
                  "text": "red",
                  "value": "#B33F62",
                },
                Object {
                  "icon": "point green",
                  "key": "green",
                  "text": "green",
                  "value": "#2db528",
                },
                Object {
                  "icon": "point pink",
                  "key": "pink",
                  "text": "pink",
                  "value": "#F851F6",
                },
                Object {
                  "icon": "point orange",
                  "key": "orange",
                  "text": "orange",
                  "value": "#e17034",
                },
                Object {
                  "icon": "point yellow",
                  "key": "yellow",
                  "text": "yellow",
                  "value": "#fdee0c",
                },
              ]
            }
          />
        </FormField>
        <FormField>
          <label>
            Occupation
          </label>
          <Styled(Input)
            name="occupation"
            onChange={[Function]}
            placeholder="Content Creator"
          />
        </FormField>
        <FormField>
          <label>
            Facebook
          </label>
          <Styled(Input)
            name="facebook_profile"
            onChange={[Function]}
            placeholder="ciroma@chukwuma.com"
          />
        </FormField>
        <FormField>
          <label>
            Twitter
          </label>
          <Styled(Input)
            name="twitter_profile"
            onChange={[Function]}
            placeholder="ciroma@chukwuma.com"
          />
        </FormField>
        <FormField>
          <label>
            LinkedIn
          </label>
          <Styled(Input)
            name="linkedin_profile"
            onChange={[Function]}
            placeholder="linkedin.com/something"
          />
        </FormField>
        <FormField>
          <label>
            Github
          </label>
          <Styled(Input)
            name="github_profile"
            onChange={[Function]}
            placeholder="github.com/you"
          />
        </FormField>
        <FormField>
          <label>
            Personal website/blog
          </label>
          <Styled(Input)
            name="site_link"
            onChange={[Function]}
            placeholder="ciroma@chukwuma.com"
          />
        </FormField>
        <Button
          className="space-top"
          disabled={true}
          inverted={true}
          type="submit"
        >
          Save
        </Button>
      </Form>
    </ContentPanel>
  </div>
</styled.div>
`;
